### ISSUE TITLE
Unwanted HTML Encoding in Source Mode for &sect

### ISSUE THREAD

**Issue by cjnething** [ckeditor/ckeditor-dev#1414]
## Are you reporting a feature request or a bug?
Bug

## [Check if the issue is already reported](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_issues_readme-section-avoid-duplicates)

Not that I could find

## Provide detailed reproduction steps (if any)

1. Visit a basic CKEditor, i.e. https://sdk.ckeditor.com/samples/standardpreset.html
2. In "source mode", put in a link with a parameter like "&section-id=12", for example:
```
<a href="http://www.test.com/?cmd=test&section-id=12">http://www.test.com/?cmd=test&section-id=12</a>
```
3. Switch back to normal (not source) mode
4. Observe that the link has changed to http://www.test.com/?cmd=test§ion-id=12

### Expected result

*What is the expected result of the above steps?*

The link should stay as http://www.test.com/?cmd=test&section-id=12

### Actual result

*What is the actual result of the above steps?*

The link changes to http://www.test.com/?cmd=test§ion-id=12

## Other details
I tried a variety of options in terms of configurations and plugins.

This was the configuration I was originally using:
```
var CKBUILDER_CONFIG = {
	skin: 'moono',
	preset: 'full',
	ignore: [
		'.DS_Store',
		'.bender',
		'.editorconfig',
		'.gitattributes',
		'.gitignore',
		'.idea',
		'.jscsrc',
		'.jshintignore',
		'.jshintrc',
		'.mailmap',
		'.travis.yml',
		'README.md',
		'bender-err.log',
		'bender-out.log',
		'bender.ci.js',
		'bender.js',
		'dev',
		'gruntfile.js',
		'less',
		'node_modules',
		'package.json',
		'tests'
	],
	plugins : {
		'a11yhelp' : 1,
		'about' : 1,
		'ajax' : 1,
		'autogrow' : 1,
		'basicstyles' : 1,
		'bbcode' : 1,
		'bidi' : 1,
		'blockquote' : 1,
		'button' : 1,
		'clipboard' : 1,
		'codemirror' : 1,
		'colorbutton' : 1,
		'colordialog' : 1,
		'confighelper' : 1,
		'contextmenu' : 1,
		'devtools' : 1,
		'dialog' : 1,
		'dialogadvtab' : 1,
		'dialogui' : 1,
		'div' : 1,
		'divarea' : 1,
		'docprops' : 1,
		'elementspath' : 1,
		'enterkey' : 1,
		'entities' : 1,
		'fakeobjects' : 1,
		'filebrowser' : 1,
		'find' : 1,
		'flash' : 1,
		'floatingspace' : 1,
		'floatpanel' : 1,
		'font' : 1,
		'format' : 1,
		'forms' : 1,
		'horizontalrule' : 1,
		'htmlbuttons' : 1,
		'htmlwriter' : 1,
		'iframe' : 1,
		'iframedialog' : 1,
		'image' : 1,
		'indent' : 1,
		'indentlist' : 1,
		'insertpre' : 1,
		'justify' : 1,
		'link' : 1,
		'list' : 1,
		'listblock' : 1,
		'liststyle' : 1,
		'magicline' : 1,
		'maximize' : 1,
		'mediaembed' : 1,
		'menu' : 1,
		'menubutton' : 1,
		'newpage' : 1,
		'oembed' : 1,
		'onchange' : 1,
		'pagebreak' : 1,
		'panel' : 1,
		'panelbutton' : 1,
		'pastefromword' : 1,
		'pastetext' : 1,
		'placeholder' : 1,
		'popup' : 1,
		'preview' : 1,
		'print' : 1,
		'removeformat' : 1,
		'resize' : 1,
		'richcombo' : 1,
		'save' : 1,
		'scayt' : 1,
		'selectall' : 1,
		'showblocks' : 1,
		'showborders' : 1,
		'smiley' : 1,
		'sourcearea' : 1,
		'specialchar' : 1,
		'stylescombo' : 1,
		'stylesheetparser' : 1,
		'symbol' : 1,
		'syntaxhighlight' : 1,
		'tab' : 1,
		'table' : 1,
		'tableresize' : 1,
		'tabletools' : 1,
		'templates' : 1,
		'toolbar' : 1,
		'uicolor' : 1,
		'undo' : 1,
		'uploadcare' : 1,
		'wordcount' : 1,
		'wsc' : 1,
		'wysiwygarea' : 1,
		'xml' : 1
	},
	languages : {
		'en' : 1
	}
};
```

In that original configuration (all of my previous plugins enabled), I tried the following configurations:
```
  config.forceSimpleAmpersand = true;
  config.htmlEncodeOutput = false;
  config.entities = false;
```

Then I tried removing as many plugins as possible, including the "entities" plugin, and created the most basic editor I could make, with the following configurations:
```
var CKBUILDER_CONFIG = {
	skin: 'moono',
	preset: 'basic',
	ignore: [
		'.DS_Store',
		'.bender',
		'.editorconfig',
		'.gitattributes',
		'.gitignore',
		'.idea',
		'.jscsrc',
		'.jshintignore',
		'.jshintrc',
		'.mailmap',
		'.travis.yml',
		'bender-err.log',
		'bender-out.log',
		'bender.ci.js',
		'bender.js',
		'dev',
		'gruntfile.js',
		'less',
		'node_modules',
		'package.json',
		'tests'
	],
	plugins : {
		'clipboard' : 1,
		'enterkey' : 1,
		'link' : 1,
		'sourcearea' : 1,
		'toolbar' : 1,
		'wysiwygarea' : 1
	},
	languages : {
		'en' : 1
	}
};
```
* Browser: Chrome
* OS: Ubuntu 17.10
* CKEditor version: 4.8
* Installed CKEditor plugins: Listed above

**Comment by mlewand** [comment]
@cjnething Thanks for a detailed issue. Your problem actually comes from a legacy behavior in browsers, that some entities will get decoded even without semicolon. You can find more explanation here:

https://stackoverflow.com/questions/18689230/why-do-html-entity-names-with-dec-255-not-require-semicolon

To avoid the problem, you want your entities to be _encoded_ when setting CKEditor HTML, so your input content should look more like:

```html
<p><a href="http://www.test.com/?cmd=test&amp;section-id=12">http://www.test.com/?cmd=test&amp;section-id=12</a></p>
```

This will prevent this legacy behavior from triggering.

Basically it's a result of upstream (browsers). It could be suggested to W3C - but I'm not sure if it will be touched there.

**Comment by cjnething** [comment]
Thanks for your help @mlewand :+1:
